assets/js/modules/tagmanager/setup.js

```diff
@@ -68,6 +76,12 @@ class TagmanagerSetup extends Component {
 		this.refetchAccount = this.refetchAccount.bind( this );
 	}
 
+	setState() {
+		if ( this._isMounted ) {
+			Component.prototype.setState.apply( this, arguments );
+		}
+	}
```

Comment:
This is quite hacky indeed. Let's instead cancel requests properly when the component is unmounted, as you already suggested.

Reply:
This may be less than ideal but it is much cleaner than before and I'm not sure if making requests cancellable is within the scope of this PR as it would likely require changes to our data API. Cancelling fetch requests requires the use of [`AbortController`](https://developer.mozilla.org/en-US/docs/Web/API/AbortController) which is technically an experimental API so I wasn't sure if we should rely on it just yet either.